ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/stats/HiveRelMdSize.java

```diff
@@ -183,4 +185,50 @@ public static Double averageTypeSize(RelDataType type) {
     }
   }
 
+  public static double typeSize(RelDataType type, Comparable value) {
```

Comment:
The method was taken from Calcite's {{RelMdSize}} as is. Interestingly, I realized that although constructor for super class is protected, I could reuse it since HiveRelMdSize is extending the class. As a bonus, I could remove `averageTypeSize` too that was there for legacy reasons and use the one in `RelMdSize`.